File: 每日一题/q35. 搜索插入位置/35.c
#include <stdio.h>
#include <stdlib.h>

int searchInsert(int *nums, int numsSize, int target)
{
    int left = 0;
    int right = numsSize - 1;
    int mid = (left + right) / 2;

    // dividing at half
    while (left <= right)
    {
        if (target < nums[mid])
        {
            right = mid - 1;
            mid = (left + right) / 2;
        }
        else if (target > nums[mid])
        {
            left = mid + 1;
            mid = (left + right) / 2;
        }
        else
            return mid;
    }
    return right + 1;
}

int main()
{
    int nums1[] = {1, 3, 5, 6};
    int numsSize1 = 4;
    int target1 = 5;
    int target2 = 2;
    int target3 = 7;
    printf("%d", searchInsert(&nums1, numsSize1, target3));
}

File: 每日一题/q35. 搜索插入位置/35.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        left = 0
        right = len(nums)-1
        while left <= right:
            mid = (int)((left+right)/2)
            if target < nums[mid]:
                right = mid-1
            elif target > nums[mid]:
                left = mid+1
            else:
                return mid
        return right+1


if __name__ == "__main__":
    nums = [1, 3, 5, 6]
    target = 5
    sol1 = Solution()
    print(sol1.searchInsert(nums, target))
